datafusion/physical-expr/src/analysis.rs

```diff
@@ -111,6 +115,22 @@ impl ExprBoundaries {
             distinct_count: col_stats.distinct_count.clone(),
         })
     }
+
+    /// Create `ExprBoundaries` that represent no known bounds for all the columns `schema`
+    pub fn try_new_unknown(schema: &Schema) -> Result<Vec<Self>> {
```

Comment:
This was added to make the demo easier to write (I ported it from upstream)

Reply:
Might unbounded be more obvious a name than unknown?

Reply:
> Might unbounded be more obvious a name than unknown?

I agree -- will change.